python/surf/devices/ti/_Ina237.py

```diff
@@ -0,0 +1,249 @@
+
+        self.add(pr.RemoteVariable(
+            name         = 'VSHUNT',
+            description  = 'Differential voltage measured across the shunt output. Twos complement value',
+            offset       = (0x4 << 2),
+            bitSize      = 16,
+            bitOffset    = 0,
+            base         = pr.Int,
+            mode         = 'RO',
+            pollInterval = pollInterval,
+            hidden       = True,
+        ))
+
+        ##############
+        # 5h VBUS
+        ##############
+
+        self.add(pr.RemoteVariable(
+            name         = 'VBUS',
+            description  = 'Bus voltage output. Twos complement value, however always positive',
+            offset       = (0x5 << 2),
+            bitSize      = 16,
+            bitOffset    = 0,
+            base         = pr.Int,
+            mode         = 'RO',
+            pollInterval = pollInterval,
+            hidden       = True,
+        ))
+
+        ##############
+        # 6h DIETEMP
+        ##############
+
+        self.add(pr.RemoteVariable(
+            name         = 'DIETEMP',
+            description  = 'Internal die temperature measurement. Twos complement value',
+            offset       = (0x6 << 2),
+            bitSize      = 12,
+            bitOffset    = 4,
+            base         = pr.Int,
+            mode         = 'RO',
+            pollInterval = pollInterval,
+            hidden       = True,
+        ))
+
+        ###############################################################################
+
+        self.add(pr.RemoteVariable(
+            name         = 'MANFID',
+            description  = 'Reads back TI in ASCII (should be 0x5449)',
+            offset       = (0x3E << 2),
+            bitSize      = 16,
+            bitOffset    = 0,
+            mode         = 'RO',
+        ))
+
+        ###############################################################################
+
+        self.add(pr.LinkVariable(
+            name         = 'Vin',
+            description  = 'Voltage Measurement',
+            mode         = 'RO',
+            linkedGet    = lambda: self.VBUS.value()*3.125E-3, # Conversion factor: 3.125 mV/LSB
+            typeStr      = "Float32",
+            disp         = '{:1.3f}',
+            units        = 'V',
+            dependencies = [self.VBUS],
+        ))
+
+        self.add(pr.LinkVariable(
+            name         = 'Iin',
+            description  = 'Current Measurement',
+            mode         = 'RO',
+            linkedGet    = self.convCurrent,
+            typeStr      = "Float32",
+            disp         = '{:1.3f}',
+            units        = 'A',
+            dependencies = [self.ADCRANGE,self.VSHUNT,self.senseRes],
+        ))
+
+        self.add(pr.LinkVariable(
+            name         = 'Pin',
+            description  = 'Power Measurement',
+            mode         = 'RO',
+            linkedGet    = lambda: (self.Vin.value())*(self.Iin.value()),
```

Comment:
Still a few .value() calls in linkedGet in this file.

Reply:
Fixed [c5eddd7](https://github.com/slaclab/surf/pull/1158/commits/c5eddd76eb186f1b7602d9b1d7eecd5234f269c7)